wordpress-mobile/gutenberg-mobile
# Upgrade react native

Requires https://github.com/wordpress-mobile/react-native-recyclerview-list/pull/10
Requires https://github.com/wordpress-mobile/react-native-aztec/pull/74

This PR upgrades react-native to `v0.57.5` as well as react to `v16.6.1` (`react-native@0.57.5`'s  peer dependency version).

It does the necessary updates in the babel config, dependencies and submodules.
It also replaces the babel transform that was importing react automatically when JSX was found with a custom one importing the gutenberg package instead (see https://github.com/wordpress-mobile/gutenberg-mobile/pull/184)

### Testing Instructions
- Check that nothing breaks:
- `yarn install` (Installing the required dependencies)
- `yarn start` (Starting the metro server)
- `yarn ios` (Running the app for ios)
- `yarn android` (Running the app for android)
- `bash ./.travis/travis-checks-js.sh` (Tests)
